content/blog/2018/egress-https/index.md

```diff
@@ -105,91 +147,194 @@ Now accessing the web page of the application displays the book details without
-microservice, when running inside a mesh, must issue HTTP requests to the port `443`, which is the port the external service listens to.
+Note that the port is derived by the `URI.parse` from the URI's schema (`https://`) to be `443`, the default HTTPS port.
+When the `DO_NOT_ENCRYPT` environment variable is defined, the request is performed without SSL (plain HTTP).
 
-When the `WITH_ISTIO` environment variable is defined, the request is performed without SSL (plain HTTP).
-
-We set the `WITH_ISTIO` environment variable to _"true"_ in the
+You can set the `DO_NOT_ENCRYPT` environment variable to _"true"_ in the
 [Kubernetes deployment spec of details v2]({{< github_file >}}/samples/bookinfo/platform/kube/bookinfo-details-v2.yaml),
 the `container` section:
 
 {{< text yaml >}}
 env:
-- name: WITH_ISTIO
+- name: DO_NOT_ENCRYPT
   value: "true"
 {{< /text >}}
 
-#### Relation to Istio mutual TLS
-
-Note that the TLS origination in this case is unrelated to [the mutual TLS](/docs/concepts/security/#mutual-tls-authentication) applied by Istio. The TLS origination for the external services will work, whether the Istio mutual TLS is enabled or not. The **mutual** TLS secures service-to-service communication **inside** the service mesh and provides each service with a strong identity. In the case of the **external services**, we have **one-way** TLS, the same mechanism used to secure communication between a web browser and a web server. TLS is applied to the communication with external services to verify the identity of the external server and to encrypt the traffic.
-
-### Malicious microservices threat
+In the next section you will configure TLS origination for accessing an external web service.
+
+## Bookinfo with TLS origination to a Google Books web service
+
+1.  Deploy a version of _details v2_ that sends an HTTP request to
+    [Google Books APIs](https://developers.google.com/books/docs/v1/getting_started). The `DO_NOT_ENCRYPT` variable
+    is set to true in
+    [`bookinfo-details-v2.yaml`]({{< github_file >}}/samples/bookinfo/platform/kube/bookinfo-details-v2.yaml).
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo-details-v2.yaml@
+    {{< /text >}}
+
+1.  Direct the traffic destined to the _details_ microservice, to _details version v2_.
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/bookinfo/networking/virtual-service-details-v2.yaml@
+    {{< /text >}}
+
+1.  Create a mesh-external service entry for `www.google.apis` and a destination rule to perform TLS origination.
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: googleapis
+    spec:
+      hosts:
+      - www.googleapis.com
+      ports:
+      - number: 443
+        name: http-port-for-tls-origination
+        protocol: HTTP
+      resolution: DNS
```

Comment:
Rather than doing this, why don't you export port 80 as well ? Then in the virtual service, you can route traffic uniformly to port 443. Doing it this way prevents the awkward http://...:443

Reply:
@rshriram Note that in the case of this particular Ruby package, it is simpler to use the port 443. It is just one conditional:

https://github.com/istio/istio/blob/master/samples/bookinfo/src/details/details.rb#L91
```ruby
unless ENV['DO_NOT_ENCRYPT'] === 'true' then
      http.use_ssl = true
end
```

The port 443 is derived automatically.